packages/explat-client/src/types.ts

```diff
@@ -21,6 +21,10 @@ export interface ExperimentAssignment {
 	 * A marker for fallback assignments - when we can't retrieve from the server.
 	 */
 	isFallbackExperimentAssignment?: boolean;
+	/**
+	 * For keeping the original experiment name in case it is invalid and we can't put it in experimentName.
+	 */
+	originalExperimentName?: string;
```

Comment:
I still don't fully understand why this is needed, but I guess that being explicit is better than changing the experiment name. Maybe we can call this `requestedExperimentName`, though?

Alternatively, we can set the type of `experimentName` to `string | null`, set it to null if the requested experiment name is invalid, and drop this field. I think it's still a bit odd to be forcing the `experimentName` to something that may one day be an actual experiment name, just to return a "valid" ExperimentAssignment under an error condition that is clearly a bug that should be caught in development (invalid experiment name).

Reply:
Hmm, yeah it's mainly from an architectural perspective I'd like all `experimentAssignments` (in general all data on the inside of a system) to be valid and consistent - even though this is a corner case. This means we don't have to have multiple `isExperimentAssignmentValid`s, ensures no future corner cases are created, and in general data is easier to reason about. For example another way to go about this might be to loosen the type to accept all strings for `experimentName`.

> Maybe we can call this requestedExperimentName, though?

👍

Reply:
> For example another way to go about this might be to loosen the type to accept all strings for `experimentName`.

Hmm... That might work better, as it'd also make it easier to change the rules around names, if we ever choose to do that.

If we do go with `requestedExperimentName`, it makes sense for it to always be non-null since _something_ was always requested.

Reply:
> Hmm... That might work better, as it'd also make it easier to change the rules around names, if we ever choose to do that.

I'm going with this, it simplifies things a lot 😄